src/reconcile.rs

```diff
@@ -285,6 +317,226 @@ where
+    ) -> ReconcileResult<Error>
+    where
+        T: HasApplication + HasRoleRestartOrder,
+        C: Command + CanBeRolling + HasRoles,
+        // TODO: Not sure if we can skip 'HasRoles' here and conditinally run code below if it is implemented
+        P: ProvidesPod,
+    {
+        // If the command provides a list of roles this overrides the default provided by the cluster
+        // definition itself
+        let role_order = command
+            .get_role_order()
+            .unwrap_or_else(|| T::get_role_restart_order());
+
+        let mut restart_occurred = false;
+        for role in role_order {
+            // Retrieve all pods for this service and role
+            let selector = self.get_role_selector(&role);
+            let pods = self
+                .client
+                .list_with_label_selector::<Pod>(self.resource.namespace().as_deref(), &selector)
+                .await?;
+
+            // Filter those out that have been restarted since the command was started
+            let pods = pods
+                .iter()
+                .filter(
+                    |pod| match (&pod.metadata.creation_timestamp, &command.start_time()) {
+                        (Some(pod_start_time), Some(command_start_time)) => {
+                            warn!(
+                                "Comparing times: [{}] < [{}]",
+                                pod_start_time.0, command_start_time
+                            );
+                            &pod_start_time.0 < command_start_time
+                                && pod.metadata.deletion_timestamp.is_none()
+                        }
+                        _ => {
+                            warn!("One of the times was not set!");
+                            false
+                        }
+                    },
+                )
+                .collect::<Vec<_>>();
+
+            if pods.is_empty() {
+                // Got no pods for this role, skip rest of processing
+                warn!(
+                    "Skipping role [{}] during restart, no pods left to restart.",
+                    role
+                );
+                continue;
+            }
+
+            // Track if anything was changed during this run
+            restart_occurred = true;
+            // Restart pods depending on strategy
+            match command.is_rolling() {
+                true => {
+                    let current_pod = pods.first().unwrap().deref();
+                    let labels = &current_pod.meta().labels;
+
+                    let role_group = labels.get(APP_ROLE_GROUP_LABEL).unwrap();
+                    let node = current_pod
+                        .spec
+                        .clone()
+                        .unwrap_or_default()
+                        .node_name
+                        .unwrap();
+
+                    self.client.ensure_deleted(current_pod.clone()).await?;
+
+                    let (pods, context) = pod_provider
```

Comment:
Maybe add a comment that this will be obsolete once https://github.com/stackabletech/operator- rs/issues/207 is implemented?